### PR TITLE
Getting rid of  remaining warnings

### PR DESCRIPTION
**Pull request by kilohsakul**
- somewhat solved few remainning warnings originating in our mata code, someone who can program should check, not too many changes
- switched of warnings from afa by pragma in aha.hh, we will probably not maintain it anyway
- have not managed to switch off warnings from re2. It should be doable through marking it as SYSTEM and for clang also adding this flag -Wno-system-headers, but does not work ... any idea?

### CODE REVIEW

**Review by tfiedor — APPROVED** [review]
No issues

**Comment by Adda0** [comment]
> have not managed to switch off warnings from re2. It should be doable through marking it as SYSTEM and for clang also adding this flag -Wno-system-headers, but does not work ... any idea?

I have tried to do the same in my PR. All tries failed miserably. I think we need to finally get into the long-awaited rewrite of our CMake in order to be able to use all CMake settings such as these. I would consider it a future improvement.

> switched of warnings from afa by pragma in aha.hh, we will probably not maintain it anyway

I considered this in my PR, but in the end decided against it. If we are not to maintain AFA any more, but want to keep the code in the repository anyway, this is probably a good option. Alternatively, we could simply stop compiling the AFA tests and source code. Depends on how likely we are to ever use this implementation of AFA in the future.

File: include/mata/util.hh [hunk 1]
```diff
@@ -580,7 +580,7 @@ void filter(Vector & vec, F && is_staying) {
 
         // remove duplicates
         auto it = std::unique(vec.begin(), vec.end());
-        vec.reserve(it - vec.begin());
+        vec.reserve(static_cast<unsigned long>(it - vec.begin()));
```
Comment (Adda0) [inline include/mata/util.hh]: I believe this should be `size_t` (which on our architectures translates to `unsigned long`). That would better explain the purpose of this cast. I will change it myself. No need to bother you with this issue.

File: src/afa/afa.cc [hunk 2]
```diff
@@ -14,6 +14,10 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
  */
+#pragma clang diagnostic push
```
Comment (Adda0) [inline src/afa/afa.cc:21]: Maybe we could add a comment above these explaining that AFA is in unmaintained state as of now.

File: tests/parser.cc [hunk 3]
```diff
@@ -417,7 +417,7 @@ TEST_CASE("correct use of Mata::Parser::parse_mf_section()")
 		REQUIRE(ref->size() == 0);
 
 		// check what remains
-		std::string remains = stream.str().substr(stream.tellg());
+		std::string remains = stream.str().substr(static_cast<unsigned long>(stream.tellg()));
```
Comment (Adda0) [inline tests/parser.cc]: This should be `size_t`. As explained above. I will make all the changes I suggested here and we can immediately merge the PR.

**Review by Adda0 — APPROVED** [review]
Seems good to me. I will resolve my issues and we can merge.

File: include/mata/util.hh (hunk 1 above)
Comment (Adda0) [reply include/mata/util.hh]: Done.

File: src/afa/afa.cc (hunk 2 above)
Comment (Adda0) [reply src/afa/afa.cc:21]: Done

File: tests/parser.cc (hunk 3 above)
Comment (Adda0) [reply tests/parser.cc]: Done.

**Pull request merged**